rholang/src/main/scala/coop/rchain/rholang/interpreter/match.scala

```diff
@@ -509,4 +792,36 @@ object SpatialMatcher {
         spatialMatch(target.source.get, pattern.source.get)
     }
 
+  // This matches a single logical connective against a Par. If it can match
+  // more ways than one, we don't care, as we care about what went into the
+  // match, and not about how it matched inside. The tricky part goes into
+  // the par/par matcher.
+  implicit val connectiveMatcher: SpatialMatcher[Par, Connective] =
+    fromFunction[Par, Connective] { (target, pattern) =>
+      pattern.connectiveInstance match {
+        case ConnAndBody(ConnectiveBody(ps)) =>
+          ps.toList.traverse_(p => spatialMatch(target, p))
+        case ConnOrBody(ConnectiveBody(ps)) => {
+          def firstMatch(target: Par, patterns: Seq[Par]): OptionalFreeMap[Unit] =
+            patterns match {
+              case Nil => StateT.pure(Unit)
+              case p +: rem =>
+                StateT[Option, FreeMap, Unit]((s: FreeMap) => {
+                  spatialMatch(target, p).run(s) match {
+                    case None            => firstMatch(target, rem).run(s)
+                    case Some((_, unit)) => Some((s, unit))
```

Comment:
I would prefer to see `Unit` (or `()`) here instead of `unit`. This is ambiguous.

Reply:
Done.